### PR TITLE
New-CmAzIaasNetworking -  allow route propagation

### LINKED ISSUE #177: Networking - UDR Propagation
We need to control UDR propagation from YML configuration files.

Basically we need to support this UI in the settings input: https://docs.microsoft.com/en-us/azure/virtual-network/tutorial-create-route-table-portal

This is a change to the route table YML and CSV.

Missed this in the networking stack review, ideally we should add this for 1.0.8.

### PR DESCRIPTION
**Pull request by deepend-dev**
resolves #177

### CODE REVIEW

File: Cloudmarque.Azure/Public/IaaS/networking/New-CmAzIaasNetworking.ps1 [hunk 1]
```diff
@@ -321,6 +321,28 @@
...
 					}
 
+					if ($routeTableGroup.routePropagation -is [array]) {
+						$routePropagation = $routeTableGroup.routePropagation[0]
+					}
+					else {
+						$routePropagation = $routeTableGroup.routePropagation
+					}
+
+					$routePropagation = switch ($routePropagation) {
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/IaaS/networking/New-CmAzIaasNetworking.ps1:331]: Is $routePropagation a string?

Comment (tm-luke) [reply Cloudmarque.Azure/Public/IaaS/networking/New-CmAzIaasNetworking.ps1:331]: Might be worth trying to cast the string value instead

**Review by tm-luke — APPROVED** [review]

File: Cloudmarque.Azure/Public/IaaS/networking/New-CmAzIaasNetworking.ps1 (hunk 1 above)
Comment (deepend-dev) [reply Cloudmarque.Azure/Public/IaaS/networking/New-CmAzIaasNetworking.ps1:331]: the problem is $routePropagation for csv is string and for yml is boolean ..

Comment (tm-luke) [reply Cloudmarque.Azure/Public/IaaS/networking/New-CmAzIaasNetworking.ps1:331]: I think .net has a tryparse method for this stuff

**Review by tm-luke — CHANGES_REQUESTED** [review]

**Review by tm-luke — APPROVED** [review]

**Pull request merged**